### PR TITLE
Issue #35 - Add ability to ignore domains

### PR DESCRIPTION
**Pull request by miketaylr**
And ignore github.com and github.io

(If we care about performance, we could probably transform the CSV stream. But since we're only looping over 500 items, this seems OK?).

### CODE REVIEW

**Comment by miketaylr** [comment]
Oh shucks, I pushed an unrelated commit before making a new branch. Let me nuke some history before we merge.

**Comment by miketaylr** [comment]
Ah, hold the presses. I'm only skipping Bugzilla here, so the webcompat results get shifted (because they don't skip).

And we're still including github.com, which we probably don't want. I guess we really want the top 502 from Tranco, and we want to ignore github.com and github.io.

**Comment by past** [comment]
Getting an arbitrary number of results from tranco is easy, you just set the `listSize` accordingly. The non-trivial part is iterating over the list and omitting the last couple of results if github.com or github.io are not on the list. 

Do we have to do the same for t.co as well? If so, what about other URL shorteners?

**Comment by miketaylr** [comment]
@past OK, I think it's ready for review now. I moved the domains into a config option and remove them from the CSV before we start fetching bugs.

As for t.co, it doesn't really hurt us to include it (since it doesn't have the same problems that github has), but it probably doesn't provide us any value. I think by default we should probably exclude it.

...we should probably hard-code these domains into config.json.example?

File: bugs.js [hunk 1]
```diff
@@ -15,6 +18,9 @@ const Octokit = require('@octokit/rest')
  *          column data
  */
 const fetchBugs = async (listFile = 'data/list.csv', bugzillaKey, githubKey, minDate, maxDate) => {
+    // These are domains we want to ignore. See https://github.com/past/tsci/issues/35
```
Comment (miketaylr) [inline bugs.js]: oops, leftover comment. will remove.

File: bugs.js [hunk 2]
```diff
@@ -26,6 +32,27 @@ const fetchBugs = async (listFile = 'data/list.csv', bugzillaKey, githubKey, min
...
+        ignoredDomains.forEach((value, index) => {
+            // create an escaped regexp out of each domain we want to ignore
+            // the CSV format will look like the following:
+            // 1,example.com\r\n
+            ignoredDomains[index] = new RegExp("\\d{1,3}," + escapeStringRegexp(value) + "\\r\\n");
+        });
+        console.log(`Skipping domains per config.ignoredDomains`);
+        const results = await replace({
+            files: listFile,
+            countMatches: true,
```
Comment (miketaylr) [inline bugs.js]: I guess I could remove this option as well, as I was just logging out the rv for debugging purposes.

File: bugs.js (hunk 1 above)
Comment (miketaylr) [reply bugs.js]: no, this ok is ok.

File: bugs.js [hunk 3]
```diff
@@ -15,6 +18,9 @@ const Octokit = require('@octokit/rest')
  *          column data
  */
 const fetchBugs = async (listFile = 'data/list.csv', bugzillaKey, githubKey, minDate, maxDate) => {
+    // These are domains we want to ignore. See https://github.com/past/tsci/issues/35
+    const ignoredDomains = config.ignoredDomains || [];
+    const ignoredDomainsCount =ignoredDomains.length;
```
Comment (wisniewskit) [inline bugs.js]: <linter_mode> nit: space after equals. </linter_mode>

File: bugs.js [hunk 4]
```diff
@@ -26,6 +32,26 @@ const fetchBugs = async (listFile = 'data/list.csv', bugzillaKey, githubKey, min
     bugTable.set("criticals", criticals);
     bugTable.set("duplicates", duplicates);
 
+    // Modify the website list, if we have any ignoredDomains.
+    if (ignoredDomainsCount) {
+        ignoredDomains.forEach((value, index) => {
+            // create an escaped regexp out of each domain we want to ignore
+            // the CSV format will look like the following:
+            // 1,example.com\r\n
+            ignoredDomains[index] = new RegExp("\\d{1,3}," + escapeStringRegexp(value) + "\\r\\n");
```
Comment (wisniewskit) [inline bugs.js]: Not a needed change, but these days I've begun to prefer using template strings:
```
const escaped = escapeStringRegexp(value);
ignoredDomains[index] = new RegExp(`\\d{1,3},${escaped}\\r\\n`);
```

File: bugs.js (hunk 3 above)
Comment (miketaylr) [reply bugs.js]: 💣 thanks.

File: bugs.js (hunk 4 above)
Comment (miketaylr) [reply bugs.js]: Yeah, this is cleaner, agreed.

**Pull request merged**